### ISSUE TITLE
Should pysand allow nan as input and return nan?

### ISSUE THREAD

**Issue by cklov** [equinor/pysand#75]
After implementing pysand in IOC erosion calculations we have encountered several issues with NaN values. IOC accept NaNs as a valid result, however pysand throws an exception if a NaN value is passed as input.  The result is that the IOC calculations fail and cannot progress past the timestamp containing a NaN and data starts to lag.

We have made some modifications to the pythonscripts in IOC to replace NaN values before they are passed to pysand, but maybe pysand should accept NaNs as this is a valid datapoint in python?

**Comment by smolvik1** [comment]
So, if I understand this correctly. You want erosion calculations to return NaN as output rather than throwing an exception when insufficient input is provided?

**Comment by cklov** [comment]
Yep. If one or more input variables is NaN the function should return NaN. I guess python by default returns NaN if NaN is part of a calculation? Any other invalid input, such as None, should probably still trigger an exception

**Comment by smolvik1** [comment]
Instinctively I am not a fan of just returning NaN on user input NaN. Would it be an option to throw a warning instead of an exception?
